app/code/core/Mage/Directory/Model/Region/Api.php

```diff
@@ -53,8 +53,8 @@ public function items($country)
 
         $result = array();
         foreach ($country->getRegions() as $region) {
-            $name = $region->getName();
-            $result[] = $region->toArray(array('region_id', 'code', $name));
+            $region->setName($region->getName());
```

Comment:
wtf is this.

Using setter to store dta from own getter function?

Reply:
The getter is overridden to return either the 'name' or the 'default_name' if the name is null. Probably at one point in time the getter would update the name itself which is why it was called without using the value, which I think is probably a worse infraction. The equivalent code for the fix without calling a setter or relying on mysterious behavior would be something like this:

```php
$regionData = $region->toArray(['region_id', 'code', 'name']);
if (empty($regionData[2])) {
    $regionData[2] = $region->getDefaultName();
}
$result[] = $regionData;
```

Or this would probably also work:

```php
$result[] = array_filter($region->toArray(['region_id', 'code', 'name', 'default_name']));
```

Reply:
@colinmollenhour good point, but to respect the original implementation of getName, the code it should look like:

```php
$regionData = $region->toArray(['region_id', 'code', 'name']);
if (is_null($regionData['name'])) {
    $regionData['name'] = $region->getData('default_name');
}
$result[] = $regionData;
```
**But it's not ideal too, the logic of the getName is cloned to other place**. At the moment, the best solution to me would be:

```php
foreach ($country->getRegions() as $region) {
    $result[] = array(
        'region_id' => $region->getRegionId(),
        'code' => $region->getCode(),
        'name' => $region->getName(), //use the logic of default name
    );
}
```